src/app-layout/visual-refresh/main.tsx

```diff
@@ -22,9 +24,13 @@ export default function Main() {
     isMobile,
     isAnyPanelOpen,
     mainElement,
+    splitPanel,
+    splitPanelReportedSize,
   } = useContext(AppLayoutContext);
 
-  const { position: splitPanelPosition } = useContext(SplitPanelContext);
+  const { getHeader: getSplitPanelHeader, position: splitPanelPosition } = useContext(SplitPanelContext);
+
+  const splitPanelReportedHeaderSize = useObservedElement(getSplitPanelHeader);
```

Comment:
Isn't this value already calculated there?
https://github.com/cloudscape-design/components/blob/main/src/app-layout/visual-refresh/layout.tsx#L71-L85

Reply:
Good catch! I with that the whole split panel height calculation can be simplified. I've updated the PR.